boilerplate/openshift/golang_osd_cluster_operator/update

```diff
@@ -17,7 +17,7 @@ THINGS YOU NEED TO DO
 =====================
 - Make sure the following line is in your base Makefile:
 
-include boilerplate/$CONVENTION_NAME/includes.mk
+include boilerplate/generated-includes.mk
```

Comment:
I'm not a big fan of a single entry point like this. You're assuming the only way to consume a conventions is through Make. I'd much rather keep the entry points in each convention subdirectory and keep the boilerplate directory clean. We then let the consumers explicitly include them.

Reply:
@dustman9000 Sorry this merged before we got a chance to finish this discussion. I'll try to summarize what we spoke about in person earlier:

> I'm not a big fan of a single entry point like this. You're assuming the only way to consume a conventions is through Make.

The only thing this assumes is that you'll consume *make targets* through Make. It doesn't affect the disposition of any boilerplate files, or any other aspect of the framework.

> I'd much rather keep the entry points in each convention subdirectory.

The advantage of the approach in this PR is that the consumer doesn't have to curate the includes in their base Makefile every time they subscribe/unsubscribe to/from a convention, or (more importantly) on updates in case the convention has changed. Yes, we have READMEs, and yes we print instructions from the per-convention `update` scripts, but this way we don't have to rely on a) not messing those up, and b) people actually reading them.

> We then let the consumers explicitly include them.

We'd like to discourage the ability for consumers to e.g. include some but not all of the `*.mk`s -- accidentally (as described above) or deliberately.

> I'd much rather {snip} keep the boilerplate directory clean.

Agreed. We discussed ways to reorganize the boilerplate directory to make it cleaner. For example:

- `boilerplate/infra/` to contain things like the `update` script, `update.cfg`, the hash of the last update, and this `generated-includes.mk` file.
- `boilerplate/conventions` to contain actual convention subdirectory structures.

That's pretty much orthogonal to this PR, but in any case something we can certainly do later.